UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/NMDataService.cs

```diff
@@ -188,31 +188,63 @@ public async Task<YearWeekResponseDataModel> GetAllYearWeek(string correlationId
 
-                BatchSearchResponse searchResult = result.Data;
+                    string searchText = $" and $batch(Frequency) eq 'Daily' and $batch(Content) eq null ";
+                    if (_userService.IsDistributorUser)
+                    {
+                        searchText = $" and $batch(Frequency) eq 'Daily' and $batch(Content) eq 'Tracings' ";
+                    }
+
+                    IFileShareApiClient fileShareApiClient = new FileShareApiClient(_httpClientFactory, _fileShareServiceConfig.Value.BaseUrl, accessToken);
+
+                    IResult<BatchSearchResponse> result = await _fileShareService.FSSBatchSearchAsync(searchText, accessToken, correlationId, fileShareApiClient);
+
+                    searchResult = result.Data;
+
+                    if (_cacheConfiguration.Value.IsFssCacheEnabled && searchResult != null && searchResult.Entries.Count > 0)
+                    {
+                        _logger.LogInformation(EventIds.FSSDailyBatchFilesResponseStoreToCacheStart.ToEventId(), "Request for storing file share service daily NM files response in azure table storage is started for with _X-Correlation-ID:{correlationId}", correlationId);
+
+                        await _fileShareServiceCache.InsertCacheObject(searchResult, rowKey, _cacheConfiguration.Value.FssCacheResponseTableName, frequency, correlationId, PartitionKey);
+
+                        _logger.LogInformation(EventIds.FSSDailyBatchFilesResponseStoreToCacheCompleted.ToEventId(), "Request for storing file share service daily NM files response in azure table storage is completed for with _X-Correlation-ID:{correlationId}", correlationId);
+                    }
+                }
 
                 if (searchResult != null && searchResult.Entries != null && searchResult.Entries.Count > 0)
                 {
                     _logger.LogInformation(EventIds.ShowDailyFilesResponseDataFound.ToEventId(), "Maritime safety information request to get daily NM files response data found for user:{SignInName} and Identity:{userId} with _X-Correlation-ID:{correlationId}", _userService.SignInName ?? "Public", _userService.UserIdentifier, correlationId);
-                    List<ShowDailyFilesResponseModel> showDailyFilesResponses = NMHelper.GetDailyShowFilesResponse(searchResult);
-                    return showDailyFilesResponses;
+
+                    ShowDailyFilesResponseListModel showDailyFilesResponseListModel = new();
+                    showDailyFilesResponseListModel.ShowDailyFilesResponseModel = NMHelper.GetDailyShowFilesResponse(searchResult);
+                    showDailyFilesResponseListModel.IsDailyFilesResponseCached = isCached;
```

Comment:
Nicer as object initialiser:
```suggestion
                    ShowDailyFilesResponseListModel showDailyFilesResponseListModel = new()
                    {
                        ShowDailyFilesResponseModel = NMHelper.GetDailyShowFilesResponse(searchResult),
                        IsDailyFilesResponseCached = isCached
                    };
```